include/shark/Models/Trees/CARTClassifier.h

```diff
@@ -100,7 +100,7 @@ class CARTClassifier : public AbstractModel<RealVector,LabelType>
 	/// Constructor taking the splitMatrix as argument
 	CARTClassifier(SplitMatrixType const& splitMatrix)
 	{
-		setSplitMatrix(splitMatrix);
+		m_splitMatrix=splitMatrix;
```

Comment:
I see. Maybe we can add a second argument "bool optimize" (i think better without default argument) instead. Then the user decides whether he wants to optimize or not.